GANDLF/models/dynunet_wrapper.py

```diff
@@ -0,0 +1,96 @@
+        dropout (Union[Tuple, str, float, None]): dropout ratio. Defaults to no dropout.
+        norm_name (Union[Tuple, str]): feature normalization type and arguments. Defaults to INSTANCE.
+        act_name (Union[Tuple, str]): activation layer type and arguments. Defaults to leakyrelu.
+        deep_supervision (bool): whether to add deep supervision head before output. Defaults to False.
+        deep_supr_num (int): number of feature maps that will output during deep supervision head. The value should be larger than 0 and less than the number of up sample layers. Defaults to 1.
+        res_block (bool): whether to use residual connection based convolution blocks during the network. Defaults to False.
+        trans_bias (bool): whether to set the bias parameter in transposed convolution layers. Defaults to False.
+    """
+
+    def __init__(self, parameters: dict):
+        super(dynunet_wrapper, self).__init__(parameters)
+
+        # checking for validation
+        assert (
+            "kernel_size" in parameters["model"]
+        ) == True, "\033[0;31m`kernel_size` key missing in parameters"
+        assert (
+            "strides" in parameters["model"]
+        ) == True, "\033[0;31m`strides` key missing in parameters"
+
+        # defining some defaults
+        if not ("upsample_kernel_size" in parameters["model"]):
+            parameters["model"]["upsample_kernel_size"] = parameters["model"][
+                "strides"
+            ][1:]
+
+        if not ("filters" in parameters["model"]):
+            parameters["model"]["filters"] = None
+
+        if not ("act_name" in parameters["model"]):
+            parameters["model"]["act_name"] = (
+                "leakyrelu",
+                {"inplace": True, "negative_slope": 0.01},
+            )
+
+        if not ("deep_supervision" in parameters["model"]):
+            parameters["model"]["deep_supervision"] = True
+
+        if not ("deep_supr_num" in parameters["model"]):
+            parameters["model"]["deep_supr_num"] = 1
+
+        if not ("res_block" in parameters["model"]):
+            parameters["model"]["res_block"] = True
+
+        if not ("trans_bias" in parameters["model"]):
+            parameters["model"]["trans_bias"] = False
+
+        if not ("norm_type" in parameters["model"]):
+            self.norm_type = "INSTANCE"
+
+        if not ("dropout" in parameters):
+            parameters["model"]["dropout"] = None
+
+        self.model = dynunet.DynUNet(
+            spatial_dims=self.n_dimensions,
+            in_channels=self.n_channels,
+            out_channels=self.base_filters,  # ? Is it correct?
+            kernel_size=parameters["model"]["kernel_size"],
+            strides=parameters["model"]["strides"],
+            upsample_kernel_size=parameters["model"][
+                "upsample_kernel_size"
+            ],  # The values should equal to strides[1:]
```

Comment:
If we already know that what the value should be, can we just hard-code that instead of getting it from the user, which could be potentially more error-prone?
```suggestion
            upsample_kernel_size=parameters["model"]["strides"][1:]
```